extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/statistics/ClusterByStatisticsCollectorImpl.java

```diff
@@ -443,7 +449,9 @@ void downSample()
       bucketHolder.keyCollector.downSample();
       newTotalRetainedBytes += bucketHolder.updateRetainedBytes();
 
-      if (i == sortedHolders.size() - 1 || sortedHolders.get(i + 1).rhs.retainedBytes > bucketHolder.retainedBytes || bucketHolder.keyCollector.estimatedRetainedKeys() <= 1) {
+      if ((bucketHolder.keyCollector.estimatedRetainedKeys() <= 1) ||
```

Comment:
Can you add the reasoning for the if..else here?

Reply:
Reverted this change